serde_derive/src/de.rs

```diff
@@ -1965,7 +1964,7 @@ fn deserialize_generated_identifier(
     quote_block! {
         #[allow(non_camel_case_types)]
         #[doc(hidden)]
-        enum __Field #lifetime {
+        enum #this_value #lifetime {
```

Comment:
I think the use of `#this_value` is semantically incorrect here and in all the places below in this function. `this_value` refers to the expression-ized (using turbofish) name of a potentially remote type. For example in the case of `serde(remote = "Foo<Bar>")`, we'd have this_type=`Foo<Bar>` and this_value=`Foo::<Bar>`. The turbofish makes it appropriate for use in expression position such as `let _ = #this_value::deserialize(d)`, as opposed to type position such as `let _: #this_value = ...`.

That is not consistent with how you've applied it in this function.

Reply:
Yes, this line appeared due to rebase from earlier implementation where there was a `#this`. It was not obvious in the first place that not just name was changed. I've changed that to use semantically correct name